lib/Cake/Model/Datasource/DboSource.php

```diff
@@ -3459,12 +3469,16 @@ public function buildColumn($column) {
 			return null;
 		}
 
-		if (!isset($this->columns[$type])) {
-			trigger_error(__d('cake_dev', 'Column type %s does not exist', $type), E_USER_WARNING);
-			return null;
+		// if a storage requirement in bytes was set, try it first (i.e. 'integer/4' before 'integer')
+		if (isset($storage) && isset($this->columns[$type . '/' . $storage])) {
+			$real = $this->columns[$type . '/' . $storage];
```

Comment:
Should we add new abstract types for the various int sizes then? Right now we're doing that implicitly with a less obvious syntax than `'type => 'smallint'`. I recognize that these new types would be less 'abstract' but there is tension between abstract types and fine grained control. Adding explicit types would let us map to existing types when smallint isn't a supported type.